content/documentation/pages/2-concepts/1-architecture.md

```diff
@@ -1,33 +1,330 @@
+
+<!--NOTE-->
+
+The Data Flow server delegates to the Skipper server to deploy long-lived applications.
+
+<!--END_NOTE-->
+
+#### Streams with Sources, Processors and Sinks
+
+Spring Cloud Stream's defines the concept of a binding interface that encapsulates in code a message exchange pattern, namely what are the applications inputs and outputs.
+Several binding interfaces are provided by Spring Cloud Stream that correspond to the follow common message exchange contracts:
+
+- `Source`: Message producer to a destination
+- `Sink`: Message consumer from a destination
+- `Processor`: The combination of a Source and a Sink. A Processor consumes message from a destination and produces messages to another destination
+
+Applications of these three types are registered with Data Flow using the the using the name `source`, `processor` and `sink` to describe the `type` of the application being registered.
+
+For example, the shell syntax for registration of a `http` source (an application that listens for http requests and sends http payload to a destination) and a `log` sink (an application that consumes from a destination and logs the received message) is
+
+```
+dataflow:>app register --name http --type source --uri maven://org.springframework.cloud.stream.app:http-source-rabbit:1.2.0.RELEASE
+Successfully registered application 'source:http'
+
+dataflow:>app register --name log --type sink --uri maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.1.0.RELEASE
+Successfully registered application 'sink:log'
+```
+
+With these names, `http` and `log`, registered with Data Flow, a stream definition can be created using the **Stream Pipeline DSL** that uses a pipes and filters syntax.
+
+```
+dataflow:>stream create --name httpStream --definition "http | log"
+```
+
+The pipe symbol in **`http | log`** represents the connection of the source output to the sink input.  
+Data Flow sets the appropriate properties when deploying the stream so that the `source` can communicate with the `sink` over the messaging middleware.
+
+#### Streams with multiple inputs and outputs
+
+Sources, Sink, and Processors all have a single output, input or both.
+This is what makes it possible for Data Flow to set application properties that pair together one output destination to another input destination.
+However, a message processing application could have more than one input or output destination.
+Spring Cloud Stream supports this by allowing the user to define a custom binding interface.
+
+To define a stream that contains an application with multiple inputs and/or you must register the application using the type `app` instead of the type `source`, `sink` or `processor`.
+The stream definition uses the **Stream Application DSL**, which replaces the single pipe symbol `|` with the double pipe symbol `||`.
```

Comment:
Again, we can cross-link to the Stream Application DSL section in the feature guide - so we can avoid duplicate content here